chia/full_node/sync_store.py

```diff
@@ -66,7 +61,7 @@ def peer_has_block(
         Adds a record that a certain peer has a block.
         """
 
-        if header_hash == self.sync_target_header_hash:
+        if self.target_peak is not None and header_hash == self.target_peak.header_hash:
```

Comment:
What happens in the `self.target_peak == None` case?
Is that always the start of chain, or can that happen when we are starting up?

Reply:
Unless i miss something, it just does the same like before, it skips. Previously, in the case where now `target_peak` is `None` we checked `header_hash == None` because we had it defaulting to `None` too, see 
https://github.com/Chia-Network/chia-blockchain/blob/d18f205bd033b3997e19393e90ab4b0cf0c63802/chia/full_node/sync_store.py#L27-L28

which fails like `self.target_peak is not None` because `header_hash` is  `bytes32`?